### DISCUSSION TITLE
v24.10.04

### DISCUSSION THREAD

**Discussion by davidrapan** [davidrapan/ha-solarman#190] [Announcements]
## What's Changed
* [feat: Improved reading/writing semaphore](https://github.com/davidrapan/ha-solarman/commit/cdcb28ed96290de9c607e46ebe87ae263b3d4762)
* [refactor: Removed name and serial from options](https://github.com/davidrapan/ha-solarman/commit/5596ad7126cdfdb8eddbe162148ebe4c81e05710)
* [feat: Add bit in dictionaries for enum](https://github.com/davidrapan/ha-solarman/commit/7648d7fdc8b79bcee9afcf9781884222c04ddeb0)

**Full Changelog**: https://github.com/davidrapan/ha-solarman/compare/v24.09.27...v24.10.04

<hr /><em>This discussion was created from the release <a href='https://github.com/davidrapan/ha-solarman/releases/tag/v24.10.04'>v24.10.04</a>.</em>

**Comment by wepster** [comment]
![image](https://github.com/user-attachments/assets/78810a6f-eaf2-484d-b70b-74d38d15d838)
How to change names of entities from inverter_ back to solarman_?
This changed in last update, and i dont like it because it deletes my history and makes me reconfigure energy and all automations

**Reply by wepster** [reply]
I know i can do it individually, but thas is a lot of work and changing device name does nothing

**Reply by davidrapan** [reply]
Entity prefix (Inverter, Solarman, ...) is selected during device configuration.

**Reply by PV-Joe** [reply]
> This changed in last update, and i dont like it because it deletes my history and makes me reconfigure energy and all automations

Just change it to Solarman.
It wasn't changed with the last update by the way.

**Reply by davidrapan** [reply]
@wepster Remove the device completely and re-add it back again and you will be fine (don't forget to set the name to required value `Solarman` during configuration).

**Reply by wepster** [reply]
I readded it already. But this shouldn't happen.

@PV-Joe:
Update was installed at 11:56:30 (first screen)
Entities went unavailable at 11:58:33 (second screen)
I think it was the update.
![image](https://github.com/user-attachments/assets/7b359d59-b1fd-410a-bb03-77f8a6c9809e)
![image](https://github.com/user-attachments/assets/b1d12299-632e-45c3-9296-94084d584b6a)

**Reply by davidrapan** [reply]
You probably changed the name of the device sometime in the past using `Configure` button from "Inverter" to "Solarman" after configuration which was broken. 😉

**Reply by wepster** [reply]
i tried this today and it didn't work, why would it work earlier? something changed?

**Reply by davidrapan** [reply]
What did you tried?

**Reply by wepster** [reply]
I tried to change name of the device (using configure button) hoping it would change prefix in entities names. But i didn't, not even after restart.
I don't remember what i did or didn't weeks ago, it is possible i messed with it.
Readding device solved my problem, but i had to again hide some entities i don't want to be easily available (for safety) - that addes some work.

Case closed.

**Reply by PV-Joe** [reply]
Congratulations 👍
Nice that it worked out for you!

**Comment by PV-Joe** [comment]
Updated today and everything went smooth without any problems for my Deye Hybrid with deye_sg04lp3.yaml 👍

Thx 4 u'r hard work enhancing this excellent integration!
